viewing_party/party.py

```diff
@@ -1,23 +1,185 @@
+        #Set the "watched" key to the list of watched movies as a value
+        user_data["watched"] = []  
+    #Add movie to the watched list
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    #Add movie to watchlist inside of user_data
+    if "watchlist" not in user_data:
+        user_data["watchlist"] = movie
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    watched_movies = user_data["watched"]
+
+    for movie in watchlist:
+        if movie["title"] == title:
+            #Remove item of type dict from watchlist of type list
+            watchlist.remove(movie)
+            watched_movies.append(movie)
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data["watched"]
+    sum_rating = 0
+    count = 0
+    average_rating = 0.0
+    
+    #In case when watched_movies is an empty list
+    if watched_movies == []:
+        return average_rating
+    # Iterate through list of dicts called watched_movies
+    else:
+        for movie in watched_movies:
+            sum_rating += movie["rating"]
+            count += 1
+    # Access the rating key in that dict
+    # Sum every rating and calculate average
+    average_rating = sum_rating/count
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    watched_movies = user_data["watched"]
+    genre_dict = {}
+
+    #In case when watched_movies is an empty list
+    if watched_movies == []:
+        return None
+    # Iterate through list of dicts called watched_movies
+    else:
+        for movie in watched_movies:
+            if movie["genre"] not in genre_dict:
+                genre_dict[movie["genre"]] = 1
+            else:
+                genre_dict[movie["genre"]] += 1
+
+    #Get the key with the maximum value 
+    most_watched_genre_count = 0
+    most_watched_genre = ""
+
+    # Iterate through list of dicts called genre_dict
+    for genre in genre_dict.keys():
+        if genre_dict[genre] > most_watched_genre_count:
+            most_watched_genre_count = genre_dict[genre]
+            most_watched_genre = genre
+    return most_watched_genre
 
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
-        
-# -----------------------------------------
+def get_unique_watched(user_data):
+    user_watched_list = user_data["watched"]
+    friends_list = user_data["friends"]
+    
+    friends_watched_set = set()
+    unique_watched = []
+
+    #Iterate through the movies in nested friends_list
+    for friend in friends_list:
+        for movie in friend["watched"]:  
+            if movie in user_watched_list:

```

Comment:
You could actually remove this line and your code would work the same. You are essentially making a flatten list of all the movies in a user's friend list. That isn't dependent on if a user has watched those movies or not.

Very nice work on the flatten list! Makes the looping so much easier!
```suggestion
```